lib/feed/view/feed_page.dart

```diff
@@ -415,6 +422,41 @@ class _FeedViewState extends State<FeedView> {
       ),
     );
   }
+
+  FutureOr<bool> _handleBack(bool stopDefaultButtonEvent, RouteInfo info) async {
+    FeedBloc feedBloc = context.read<FeedBloc>();
+
+    // See if we're at the top level of navigation
+    final canPop = Navigator.of(context).canPop();
+
+    // Get the desired post listing so we can check against current
+    final prefs = (await UserPreferences.instance).sharedPreferences;
+    final desiredPostListingType = PostListingType.values.byName(prefs.getString("setting_general_default_listing_type") ?? DEFAULT_LISTING_TYPE.name);
+    final currentPostListingType = feedBloc.state.postListingType;
+
+    // See if we're in a community
+    final currentCommunityId = feedBloc.state.communityId;
```

Comment:
For this, I would suggest using `feedBloc.state.feedType == FeedType.community` since that's generally what is used elsewhere when checking the type of feed (general, community, and eventually user)!

Reply:
That's much nicer! Done!